django/contrib/admin/static/admin/css/nav_sidebar.css

```diff
@@ -40,6 +40,7 @@
     border-right: 1px solid var(--hairline-color);
     background-color: var(--body-bg);
     overflow: auto;
+    height: calc(100vh - 87px);
```

Comment:
Thanks for tackling this :)

It's always a bit unfortunate to have to hard-code values like this, but sometimes it _is_ a necessity. That said, why **87px**? From my brief playing with the sizing (on 4.0, tbf) in Chrome on OSX, my body scrollbar disappears at **85px** (which is presumably header + breadcrumbs + some border on one or the other), which makes me wonder about it.

Is there a reason why the property change lives here on `#nav-sidebar`? It seems like it _could_ perhaps live on `.sticky` as `max-height: calc(100vh - 8Xpx)` ... of course I could be missing an edge case, CSS being what it is.

Having these questions would be reason to _me_ to have comments explaining why (because if I'm tweaking it downstream in my project, I'd like to know _why_ I have to), in the CSS, but that doesn't seem particularly prevalent in the existing CSS files.

Reply:
I chose 87px because on my Windows machine, in Chrome, the header has a height of 49.6px and the breadcrumbs have a height of 36.8px. So I chose 50 + 37 = 87px.

As for why the property lives on `#nav-sidebar` and not another selector - this is because this height change would be specific to the navigation sidebar. There is also a button beside the `#nav-sidebar` that has the `.sticky` class that collapses and expands the sidebar, and if the property was on `sticky`, that button's height would also have been reduced. That may be okay too, but in my opinion, leaving the button with a height of 100vh feels better and exposes a larger area to click on when the page is scrolled. No other logic there.

However, `.sticky` seems like a utility class that might be used to give other elements a sticky positioning in the future, and if we give that a height of `calc(100vh - 87px)` or `calc(100vh - 119px)` after the breakpoint, it may lead to undesirable behavior.

Reply:
> I chose 87px because on my Windows machine, in Chrome, the header has a height of 49.6px and the breadcrumbs have a height of 36.8px. So I chose 50 + 37 = 87px.

Cool, that's a useful explanation, and is kind of the reason I was wondering about how you derived that value (my values don't appear to be fractional, so mine works at a lower value).

> As for why the property lives on #nav-sidebar and not another selector[...] that button's height would also have been reduced. 

Semantics of where it lives notwithstanding (_I'm_ not fussed), I really just wanted to know _why there_. I expect both places are fine in practice, but given the way CSS cascades (that you can't infer from a diff) there could've been a hard _requirement_ that it live on `#nav-sidebar` rather than anywhere else.
I've not had much cause to use the admin since the arrival of the sidebar, but I'm not sure I follow why the collapsing button would be noticeably affected - aren't they intentionally _both_ sticky & the same height? (Caveat emptor myself: no there appears to be a **1px** difference for me, because neither use `box-sizing` ... odd)

_Personally_, I'd ignore the "what could be" of the `.sticky` class usage, and put the value there so that downstream modifications only need to account for one override/tweak, but that ship may have sailed given `.sticky` is already "in the wild". Mostly I just wanted to explore the whys and wherefores of the changes, so thanks for having indulged my questions :)

Reply:
Thank you :)

There is no hard requirement that the property must be given to `#nav-sidebar`, but it does result in the collapsing button spanning the entire height of the screen. Intuitively, both options (having the button span the entire height as well as shortening the button) seem just a little off, as in both cases we either get a mismatch in the heights of the button and the nav, or both the nav and the button are shortened, leaving a small blank space when the page is scrolled.

![image](https://user-images.githubusercontent.com/64729027/147376136-a1703592-ade2-46a7-a12b-79ad2cba9a2f.png)

To be honest, attaching the property to `.sticky` seems odd to me because I would generally prefer to give a property as specific as this to the most specific selector targeting our element of concern to prevent side effects, but I see how in this case `.sticky` would be okay as well, since it already has a specific height given to it and it hasn't been used anywhere else in the admin HTML as far as I can tell. So I guess both places would be fine :)